### ISSUE TITLE
intra_s3 route table association: element: element() may not be used with an empty list

### ISSUE THREAD

**Issue by mmarod** [terraform-aws-modules/terraform-aws-vpc#137]
Running into an issue with aws_vpc_endpoint_route_table_association.intra_s3 since https://github.com/terraform-aws-modules/terraform-aws-vpc/pull/135 was merged.

```
Error: Error running plan: 1 error(s) occurred:

* module.network.aws_vpc_endpoint_route_table_association.intra_s3: 1 error(s) occurred:

* module.network.aws_vpc_endpoint_route_table_association.intra_s3: element: element() may not be used with an empty list in:

${element(aws_route_table.intra.*.id, 0)}
```

**Comment by antonbabenko** [comment]
Please provide terraform configurations which leads to this error.

**Comment by mmarod** [comment]
```
module "network" {
  source = "terraform-aws-modules/vpc/aws"

  name = "${var.name}"
  cidr = "${var.cidr_block}"

  azs            = "${var.zones}"
  public_subnets = "${var.public_subnet_cidrs}"

  map_public_ip_on_launch = false
  enable_dns_hostnames    = true
  enable_dns_support      = true
  enable_s3_endpoint      = true

  tags = "${merge(local.default_tags,local.network_tags, var.tags)}"
}
```

**Comment by antonbabenko** [comment]
Fixed in [v1.34.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v1.34.0). Thank you for reporting!
